paimon-core/src/main/java/org/apache/paimon/operation/PartitionExpire.java

```diff
@@ -75,7 +76,13 @@ public PartitionExpire(
         this.scan = scan;
         this.commit = commit;
         this.partitionHandler = partitionHandler;
-        this.lastCheck = LocalDateTime.now();
+        // Avoid the execution time of stream jobs from being too short and preventing partition
+        // expiration
+        long rndSeconds = 0;
+        if (!checkInterval.isZero()) {
+            rndSeconds = ThreadLocalRandom.current().nextLong(checkInterval.toMillis() / 1000);
```

Comment:
extreme case: `0 < checkInterval < 1000`.

Reply:
I will fix it.